utilities/AFQ_mrtrixInit.m

```diff
@@ -80,101 +126,154 @@
+                      false, ... % this is show figure
+                      [], ... %bckground
+                      8, ... %lmax
+                      false, ... %verbose
+                      mrtrixVersion) 
 end
 
 % Create a white-matter mask, tracktography will act only in here.
-if (~computed.('wm'))
+if (~computed.('wmMask'))
   wmMaskFile = fullfile(session, dt_info.files.wmMask);
-  mrtrix_mrconvert(wmMaskFile, files.wm,[],0)
+  AFQ_mrtrix_mrconvert(wmMaskFile, ...
+                       files.wmMask, ...
+                       [], ...
+                       0, ...
+                       mrtrixVersion)
 end
 
-% Compute the CSD estimates: 
-if (~computed.('csd'))  
-  disp('The following step takes a while (a few hours)');                                  
-  mrtrix_csdeconv(files.dwi, files.response, lmax, files.csd, files.b, files.brainmask,0)
-end
-
-
 
-%%%%%%%%%%%%%%%%%%%%%%%%%%
-% mrtrix_check_processes %
-%%%%%%%%%%%%%%%%%%%%%%%%%%
-function computed = mrtrix_check_processes(files)
-%
-% Check which mrtrix proceses were computed. Returns an array of 0's and
-% 1's indicating which process was computed (1) and which one needs to be
-% computed (0)
-%
 
-fields = fieldnames(files);
-for ii = 1:length(fields)
-  if exist(files.(fields{ii}),'file') == 2
-    computed.(fields{ii}) = 1;
-  else
-    computed.(fields{ii}) = 0;
-  end
+if ~multishell
+    % Compute the CSD estimates: 
+    if (~computed.('csd'))  
+      disp('The following step takes a while (a few hours)');                                  
+      AFQ_mrtrix_csdeconv(files.dwi, ...
+                          files.response, ...
+                          lmax, ...
+                          files.csd, ... %out
+                          files.b, ... %grad
+                          files.brainmask,... %mask
+                          false,... % Verbose
+                          mrtrixVersion)
+    end
+
+else
+    % Create the 5tt file from the same ac-pc-ed T1 nii we used in the other steps: 
+    if (~computed.('tt5')) && (mrtrixVersion > 2)
+        inputFile = [];
+        if strcmp(tool, 'fsl')
+            inputFile = fullfile(session, dt_info.files.t1);
+            if ~(exist(inputFile, 'file') == 2)
+                error(['Cannot find T1, please copy it to ' session]);
+            end    
+        % Find and aseg file and better if it is an aparc+aseg one. 
+        % Select the first aparc if there are several *aseg* files.
+        % It can take mgz or nii or mif
+        else
+           asegFiles = dir(fullfile(session,'*aseg*'));
+           for ii = 1:length(asegFiles)
+               if length(strfind(asegFiles(ii).name, 'aseg')) > 0
+                   inputFile = asegFiles(ii).name;
+               end
+               if length(strfind(asegFiles(ii).name, 'aparc')) > 0
+                   inputFile = asegFiles(ii).name;
+               end
+           end
+            if ~(exist(inputFile, 'file') == 2)
```

Comment:
ok! but it is working for you then? Should I check sthg else? I can do it
in 3-4 for hours, I am leaving right now. Thanks for the testing!

On Fri, Jul 15, 2016 at 8:11 PM dstrodtman <notifications@github.com> wrote:

> In utilities/AFQ_mrtrixInit.m
> <https://github.com/yeatmanlab/AFQ/pull/11#discussion_r71017035>:
>
> > +                error(['Cannot find T1, please copy it to ' session]);
> > +            end
> > +        % Find and aseg file and better if it is an aparc+aseg one.
> > +        % Select the first aparc if there are several *aseg* files.
> > +        % It can take mgz or nii or mif
> > +        else
> > +           asegFiles = dir(fullfile(session,'*aseg*'));
> > +           for ii = 1:length(asegFiles)
> > +               if length(strfind(asegFiles(ii).
>
> name, 'aseg')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +               if length(strfind(asegFiles(ii).name, 'aparc')) > 0
> > +                   inputFile = asegFiles(ii).name;
> > +               end
> > +           end
> > +            if ~(exist(inputFile, 'file') == 2)
>
> Yes, session at this point has an absolute path. I call 'sub_dirs' with a
> list of full paths to my dti*trilin directory for each subject when I
> AFQ_Create.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeatmanlab/AFQ/pull/11/files/e6e990255c225454a1920d786c05a6a68c920d07#r71017035>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEWQV-3oL-v_qTww-BDiUvqE8jE_FCs5ks5qV81qgaJpZM4JGbEU>
> .
>